packages/kbn-apm-config-loader/src/config.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { join } from 'path';
+import { merge, get } from 'lodash';
+import { execSync } from 'child_process';
+// deep import to avoid loading the whole package
+import { getDataPath } from '@kbn/utils/target/path';
+import { readFileSync } from 'fs';
+import { ApmAgentConfig } from './types';
+
+const getDefaultConfig = (isDistributable: boolean): ApmAgentConfig => {
+  if (isDistributable) {
+    return {
+      active: false,
+      globalLabels: {},
+    };
+  }
+  return {
+    active: false,
+    serverUrl: 'https://f1542b814f674090afd914960583265f.apm.us-central1.gcp.cloud.es.io:443',
+    // The secretToken below is intended to be hardcoded in this file even though
+    // it makes it public. This is not a security/privacy issue. Normally we'd
+    // instead disable the need for a secretToken in the APM Server config where
+    // the data is transmitted to, but due to how it's being hosted, it's easier,
+    // for now, to simply leave it in.
+    secretToken: 'R0Gjg46pE9K9wGestd',
+    globalLabels: {},
+    breakdownMetrics: true,
+    centralConfig: false,
+    logUncaughtExceptions: true,
+  };
+};
+
+export class ApmConfiguration {
+  private baseConfig?: any;
+  private kibanaVersion: string;
+
+  constructor(
+    private readonly rootDir: string,
+    private readonly rawKibanaConfig: Record<string, any>,
+    private readonly isDistributable: boolean
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { version } = require(join(this.rootDir, 'package.json'));
+    this.kibanaVersion = version.replace(/\./g, '_');
+  }
+
+  public getConfig(serviceName: string): ApmAgentConfig {
+    return {
+      ...this.getBaseConfig(),
+      serviceName: `${serviceName}-${this.kibanaVersion}`,
```

Comment:
I know this is how it works in the current implementation, but I think we should be using the `serviceVersion` property rather than appending the Kibana version to the `serviceName`. I also wonder if it would be useful to include the git revision, something like:

```suggestion
      serviceName,
      serviceVersion: `${this.kibanaVersion}-${this.git_rev}`
```

Reply:
Seems better yea, but I'm not sure who's in charge of confirming that the change would be alright? Is that on us?

Reply:
As of you edit, 

> I think we should merge this without any changes to the APM config right now

I will keep this for a follow up to avoid changing anything from the apm config in the current PR.

Reply:
I think that the reason why I originally went with this approach was because the APM UI didn't have a way to filter on `serviceVersion` - so it would bundle up all traces for Kibana v8.0.0, v7.9.2 etc etc. into the same view. I'm not sure if it's possible to apply such a filter easily today that will stick between clicks to different pages?

Reply:
It seems the service version filter does stick around in the APM UI when switching between different views now. It'd be nice to use the same serviceName for all versions so that we can do comparisons more easily